www/js/diary/timelineHelper.ts

```diff
@@ -113,3 +152,56 @@ export function getAllUnprocessedInputs(pipelineRange, labelsFactory, notesFacto
   );
   return getUnprocessedResults(labelsFactory, notesFactory, labelsPromises, notesPromises);
 }
+
+/**
+ * @param locationPoint an object containing coordinates as array of [lat, lon]
+ * @param featureType a string describing the feature, e.g. "start_place"
+ * @returns a GeoJSON feature with type "Point", the given location's coordinates and the given feature type
+ */
+const location2GeojsonPoint = (locationPoint: any, featureType: string) => ({
+  type: "Feature",
+  geometry: {
+    type: "Point",
+    coordinates: locationPoint.coordinates,
+  },
+  properties: {
+    feature_type: featureType,
+  }
+});
+
+/**
+ * @param trip
+ * @param locationList an array of locations to use for the trajectory.
+ * @param trajectoryColor The color to use for the whole trajectory, if any. Otherwise, a color will be lookup up for the sensed mode of each section.
+ * @returns for each section of the trip, a GeoJSON feature with type "LineString" and an array of coordinates.
+ */
+const locations2GeojsonTrajectory = (trip, locationList, trajectoryColor?) => {
+  let sectionsPoints;
+  if (!trip.sections) {
+    // this is a unimodal trip so we put all the locations in one section
+    sectionsPoints = [locationList];
+  } else {
+    // this is a multimodal trip so we sort the locations into sections by timestamp
+    sectionsPoints = trip.sections.map((s) =>
+      trip.locations.filter((l) =>
+        l.ts >= s.start_ts && l.ts <= s.end_ts
+      )
+    );
```

Comment:
I'm confused. I see from the demo that when there is a user label, we essentially display the trip as unimodal.
However, in that case, we must be setting the `trip.sections` to `[]` somewhere so that we don't just fall through and execute the multi-modal trip use case. But I don't see where that is.

Reply:
`trip.sections` is never modified because we want to treat the trip object as immutable.

If a trip has been labeled with a mode, (ie `trip.userInput.MODE.value` exists), we keep `trip.sections` as it is and just don't use it.
That discrimination happens a few places including `ModesIndicator` and `TripSectionsDescriptives`